src/Microsoft.ML.Data/DataLoadSave/Database/DatabaseSource.cs

```diff
@@ -14,7 +14,8 @@ public sealed class DatabaseSource
         /// <param name="providerFactory">The factory used to create the <see cref="DbConnection"/>..</param>
         /// <param name="connectionString">The string used to open the connection.</param>
         /// <param name="commandText">The text command to run against the data source.</param>
-        public DatabaseSource(DbProviderFactory providerFactory, string connectionString, string commandText)
+        /// <param name="commandTimeout">The time in seconds to wait for the command to execute. Default is 30 sec.</param>
+        public DatabaseSource(DbProviderFactory providerFactory, string connectionString, string commandText, int commandTimeout = 30)
```

Comment:
It's likely worth making it explicitly clear that this is in seconds.

Reply:
The XML description of the parameter does state that the time is in seconds already 

---
In reply to: [349254631](https://github.com/dotnet/machinelearning/pull/4494#discussion_r349254631) [](ancestors = 349254631)

Reply:
If I may, as a bit of an outsider, for a reason or another it's not always easy to see XML comments. So being explicit like `commandTimeoutInSeconds = 30` is easier to catch. I believe it's sort of a pattern used quite commonly too.

Reply:
Wouldn't matching the name of the actual property that is being set be more consistent. See [https://docs.microsoft.com/en-us/dotnet/api/system.data.sqlclient.sqlcommand.commandtimeout?view=netcore-3.0](https://docs.microsoft.com/en-us/dotnet/api/system.data.sqlclient.sqlcommand.commandtimeout?view=netcore-3.0). 

---
In reply to: [351692404](https://github.com/dotnet/machinelearning/pull/4494#discussion_r351692404) [](ancestors = 351692404)

Reply:
The ML API here is not, strictly speaking, tied to SQL. It is meant to be usable with any database provider. It is more "tied" to `DbCommand.CommandTimeout`: https://docs.microsoft.com/en-us/dotnet/api/system.data.common.dbcommand.commandtimeout?view=netframework-4.8#System_Data_Common_DbCommand_CommandTimeout

Adding `InSeconds` just helps clarify the unit of measurement that the timeout is (which can be particularly helpful as .NET APIs frequently deal in milliseconds instead).

Reply:
changing.

---
In reply to: [352740161](https://github.com/dotnet/machinelearning/pull/4494#discussion_r352740161) [](ancestors = 352740161)